Analysis/Tasks/PWGCF/FemtoDream/include/FemtoDream/FemtoDreamContainer.h

```diff
@@ -60,80 +64,43 @@ class FemtoDreamContainer
   {
     float femtoObs;
     if (mFemtoObs == femtoDreamContainer::Observable::kstar) {
-      femtoObs = getkstar(part1, mMassOne, part2, mMassTwo);
+      femtoObs = mFemtoDreamMath->getkstar(part1, mMassOne, part2, mMassTwo);
     }
-    const float kT = getkT(part1, mMassOne, part2, mMassTwo);
+    const float kT = FemtoDreamMath::getkT(part1, mMassOne, part2, mMassTwo);
+    const float mT = FemtoDreamMath::getmT(part1, mMassOne, part2, mMassTwo);
 
     if (mHistogramRegistry) {
       mHistogramRegistry->fill(HIST("relPairDist"), femtoObs);
       mHistogramRegistry->fill(HIST("relPairkT"), kT);
+      mHistogramRegistry->fill(HIST("relPairkstarkT"), femtoObs, kT);
+      mHistogramRegistry->fill(HIST("relPairkstarmT"), femtoObs, mT);
     }
   }
 
   template <typename T>
-  static float getkstar(const T& part1, const float mass1, const T& part2, const float mass2)
+  void SetEventPairHistograms(float multEvt, T const& part1, T const& part2)
   {
-    TLorentzVector vecpart1, vecpart2;
-    vecpart1.SetXYZM(part1.pt(), part1.eta(), part1.phi(), mass1);
-    vecpart2.SetXYZM(part2.pt(), part2.eta(), part2.phi(), mass2);
-    return getkstar(vecpart1, vecpart2);
-    //    return getkstar({part1.pt(), part1.eta(), part1.phi(), mass1}, {part2.pt(), part2.eta(), part2.phi(), mass2});
-  }
-
-  /// \todo improve performance of the computation by replacing TLorentzVector
-  //  static float getkstar(const ROOT::Math::PtEtaPhiMVector& part1, const ROOT::Math::PtEtaPhiMVector& part2);
-  static float getkstar(const TLorentzVector& part1, const TLorentzVector& part2);
+    float femtoObs;
+    if (mFemtoObs == femtoDreamContainer::Observable::kstar) {
+      femtoObs = mFemtoDreamMath->getkstar(part1, mMassOne, part2, mMassTwo);
+    }
+    const float kT = FemtoDreamMath::getkT(part1, mMassOne, part2, mMassTwo);
+    const float mT = FemtoDreamMath::getmT(part1, mMassOne, part2, mMassTwo);
 
-  /// constructor for kT calculation:
-  /// implemented to have as arguments some particle objects T or TLorenzVector
-  template <typename T>
-  static float getkT(const T& part1, const float mass1, const T& part2, const float mass2)
-  {
-    TLorentzVector vecpart1, vecpart2;
-    vecpart1.SetXYZM(part1.pt(), part1.eta(), part1.phi(), mass1);
-    vecpart2.SetXYZM(part2.pt(), part2.eta(), part2.phi(), mass2);
-    return getkT(vecpart1, vecpart2);
+    if (mHistogramRegistry) {
+      mHistogramRegistry->fill(HIST("relPairkstarMult"), femtoObs, multEvt);
```

Comment:
hmm I guess we also need the multiplicity bins according to how the mixing is done, this should also be set somewhere, and then we need to fill this histogram accordingly